### ISSUE TITLE
Proposed rule: Use delegation of mutable state rather than variable state value

### ISSUE THREAD

**Issue by VirtualParticle** [slackhq/compose-lints#375]
Storing the actual state object in a variable like this:
```kotlin
var x = remember { mutableStateOf(0) }
```
should be avoided in favor of
```kotlin
var x by remember { mutableStateOf(0) }
```
In the first example, there's no reason that the field `x` should actually be mutable since it's the `value` property within it that should be changed. Even in this example:
```kotlin
val x = remember { mutableStateOf(0) }
```
where `x` is properly immutable, it's still probably preferred that the delegation syntax be used just because it's cleaner. I and the official compose docs seem to mostly prefer the delegation syntax, but not everyone might agree.
